### PR TITLE
feat(protocol-kit): Add AbiType for Multisend

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves
Resolves https://github.com/safe-global/safe-core-sdk/issues/654

## How this PR fixes it
Migrating MultiSend and MultiSendCallOnly contracts from Typechain to Abitype

### CODE REVIEW

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendBaseContractEthers.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { Contract, InterfaceAbi } from 'ethers'
+
+import EthersAdapter from '@safe-global/protocol-kit/adapters/ethers/EthersAdapter'
+import { SafeVersion } from '@safe-global/safe-core-sdk-types'
+import MultiSendBaseContract from '@safe-global/protocol-kit/adapters/MultiSendBaseContract'
+
+/**
+ * Abstract class SafeBaseContractEthers extends MultiSendBaseContract to specifically integrate with the Ethers.js v6 library.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendBaseContractEthers.ts:8]:
```suggestion
 * Abstract class MultiSendBaseContractEthers extends MultiSendBaseContract to specifically integrate with the Ethers.js v6 library.
```

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendBaseContractEthers.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * It is designed to be instantiated for different versions of the Safe contract.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendBaseContractEthers.ts:9]:
```suggestion
 * It is designed to be instantiated for different versions of the MultiSend contract.
```

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendBaseContractEthers.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ */
+abstract class MultiSendBaseContractEthers<
+  MultiSendContractAbiType extends InterfaceAbi
+> extends MultiSendBaseContract<MultiSendContractAbiType> {
+  contract: Contract
+  adapter: EthersAdapter
+
+  /**
+   * @constructor
+   * Constructs an instance of SafeBaseContractEthers.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendBaseContractEthers.ts:30]:
```suggestion
   * Constructs an instance of MultiSendBaseContractEthers.
```

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendCallOnlyBaseContractEthers.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { Contract, InterfaceAbi } from 'ethers'
+
+import EthersAdapter from '@safe-global/protocol-kit/adapters/ethers/EthersAdapter'
+import { SafeVersion } from '@safe-global/safe-core-sdk-types'
+import MultiSendCallOnlyBaseContract from '@safe-global/protocol-kit/adapters/MultiSendCallOnlyBaseContract'
+
+/**
+ * Abstract class SafeBaseContractEthers extends MultiSendCallOnlyBaseContract to specifically integrate with the Ethers.js v6 library.
+ * It is designed to be instantiated for different versions of the Safe contract.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendCallOnlyBaseContractEthers.ts:9]:
```suggestion
 * It is designed to be instantiated for different versions of the MultiSendCallOnly contract.
```

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendCallOnlyBaseContractEthers.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+ */
+abstract class MultiSendCallOnlyBaseContractEthers<
+  MultiSendCallOnlyContractAbiType extends InterfaceAbi
+> extends MultiSendCallOnlyBaseContract<MultiSendCallOnlyContractAbiType> {
+  contract: Contract
+  adapter: EthersAdapter
+
+  /**
+   * @constructor
+   * Constructs an instance of SafeBaseContractEthers.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/MultiSendCallOnlyBaseContractEthers.ts:30]:
```suggestion
   * Constructs an instance of MultiSendCallOnlyBaseContractEthers.
```

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/v1.1.1/MultiSendContract_V1_1_1_Ethers.ts [hunk 6]
```diff
@@ -1,10 +1,59 @@
...
+  GetAddressMultiSendFunction
+} from '@safe-global/protocol-kit/contracts/AbiType/MultiSend/MultiSendBaseContract'
 
-class MultiSendContract_V1_1_1_Ethers extends MultiSendEthersContract {
-  constructor(public contract: MultiSend) {
-    super(contract)
+/**
+ * MultiSendContract_v1_1_1_Ethers is the implementation specific to the Safe contract version 1.1.1.
+ *
+ * This class specializes in handling interactions with the Safe contract version 1.1.1 using Ethers.js v6.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/v1.1.1/MultiSendContract_V1_1_1_Ethers.ts:16]:
```suggestion
 * This class specializes in handling interactions with the MultiSend contract version 1.1.1 using Ethers.js v6.
```

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/v1.1.1/MultiSendContract_V1_1_1_Ethers.ts [hunk 7]
```diff
@@ -1,10 +1,59 @@
...
+ *
+ * @extends MultiSendBaseContractEthers<MultiSendContract_v1_1_1_Abi> - Inherits from SafeBaseContractEthers with ABI specific to Safe contract version 1.1.1.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/v1.1.1/MultiSendContract_V1_1_1_Ethers.ts:18]:
```suggestion
 * @extends MultiSendBaseContractEthers<MultiSendContract_v1_1_1_Abi> - Inherits from  MultiSendBaseContractEthers with ABI specific to Safe contract version 1.1.1.
```

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/v1.1.1/MultiSendContract_V1_1_1_Ethers.ts [hunk 8]
```diff
@@ -1,10 +1,59 @@
...
+ * @implements SafeContract_v1_1_1_Contract - Implements the interface specific to Safe contract version 1.1.1.
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/v1.1.1/MultiSendContract_V1_1_1_Ethers.ts:19]:
```suggestion
 * @implements MultiSendContract_v1_1_1_Contract - Implements the interface specific to Safe contract version 1.1.1.
```

File: packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/v1.1.1/MultiSendContract_V1_1_1_Ethers.ts [hunk 9]
```diff
@@ -1,10 +1,59 @@
...
+ */
+class MultiSendContract_v1_1_1_Ethers
+  extends MultiSendBaseContractEthers<MultiSendContract_v1_1_1_Abi>
+  implements MultiSendContract_v1_1_1_Contract
+{
+  safeVersion: SafeVersion
+
+  /**
+   * Constructs an instance of SafeContract_v1_1_1_Ethers
```
Comment (DaniSomoza) [inline packages/protocol-kit/src/adapters/ethers/contracts/MultiSend/v1.1.1/MultiSendContract_V1_1_1_Ethers.ts:28]:
```suggestion
   * Constructs an instance of MultiSendContract_v1_1_1_Ethers
```

**Comment by yagopv** [comment]
All changed @DaniSomoza !

**Review by DaniSomoza — APPROVED** [review]

**Pull request merged**